junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/discovery/predicates/IsTestableMethod.java

```diff
@@ -46,6 +49,10 @@ public boolean test(Method candidate) {
 		if (isAbstract(candidate)) {
 			return false;
 		}
+		if (returnsVoid(candidate) != mustReturnVoid) {
```

Comment:
In essence, the new behavior creates false-positives when we do have a void return type but need a collection or stream return type for factory methods. So will revert to the original behavior.